streaming/base/shared.py

```diff
@@ -268,3 +272,79 @@ def cleanup(self):
             pass
         finally:
             resource_tracker.unregister = original_rtracker_unreg
+
+
+def get_shm_prefix(my_locals: List[str], world: World) -> Tuple[str, SharedMemory]:
+    """Register or lookup our shared memory prefix.
+
+    Args:
+        my_locals (List[str]): Local working dir of each stream, relative to /. We need to verify
+            that there is no overlap with any other currently running StreamingDataset.
+        world (World): Our place in the world.
+
+    Returns:
+        Tuple[str, CreateSharedMemory]: Shared memory prefix and object. The name is required to be
+            very short for Mac OSX.
+    """
+    # Local leader goes first, checking and registering.
+    if world.is_local_leader:
+        # Local leader walks the existing shm prefixes starting from zero, verifying that there is
+        # no local working directory overlap.
+        for prefix_int in range(10**6):
+            prefix = f'{prefix_int:06}'
+            name = f'{prefix}_locals'
+            try:
+                shm = CreateSharedMemory(name, False).shm
```

Comment:
Can you add a comment here saying if the shared_memory doesn't exist then it will throw FileNotFoundError which means we found the unique shared_memory prefix ?

Reply:
how about this?